dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -0,0 +1,543 @@
+                    ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } // save
+
+    /**
+     * Saves an action into a step
+     * @param request                   HttpServletRequest
+     * @param workflowActionStepForm    WorkflowActionStepForm
+     * @return Response
+     */
+    @POST
+    @Path("/steps/actions")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response saveActionToStep(@Context final HttpServletRequest request,
+                               final WorkflowActionStepForm workflowActionStepForm) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+
+        try {
+
+            Logger.debug(this, "Saving a workflow action " + workflowActionStepForm.getActionId()
+                    + " in to a step: " + workflowActionStepForm.getStepId());
+            this.workflowHelper.saveActionToStep(workflowActionStepForm, initDataObject.getUser());
+            response  = Response.ok(new ResponseEntityView("ok")).build(); // 200
+        } catch (Exception e) {
+
+            Logger.error(this.getClass(),
+                    "Exception on saveAction, workflowActionForm: " + workflowActionStepForm +
+                            ", exception message: " + e.getMessage(), e);
+            response = (e.getCause() instanceof SecurityException)?
+                    ExceptionMapperUtil.createResponse(e, Response.Status.UNAUTHORIZED) :
+                    ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } // saveAction
+
+
+    /**
+     * Deletes a step
+     * @param request                   HttpServletRequest
+     * @param stepId                   String
+     * @return Response
+     */
+    @DELETE
+    @Path("/steps/{stepId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response deleteStep(@Context final HttpServletRequest request,
+                                           @PathParam("stepId") final String stepId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+
+        try {
+
+            Logger.debug(this, "Deleting the step: " + stepId);
+            this.workflowHelper.deleteStep(stepId);
+            response  = Response.ok(new ResponseEntityView("ok")).build(); // 200
+        } catch (DoesNotExistException e) {
+
+            Logger.error(this.getClass(),
+                    "DoesNotExistException on deleteStep, stepId: " + stepId +
+                            ", exception message: " + e.getMessage(), e);
+            response = ExceptionMapperUtil.createResponse(e, Response.Status.NOT_FOUND);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)